addons/mail/static/src/new/messaging.js

```diff
@@ -153,8 +153,9 @@ export class Messaging {
     /**
      * todo: merge this with Thread.insert() (?)
      */
-    createChannelThread(serverData) {
+    async createChannelThread(serverData) {
         const { id, name, last_message_id, seen_message_id, description, channel } = serverData;
+        const channelMembers = await this.getChannelMembers(channel);
```

Comment:
You should not fetch `ChannelMembers` immediately whenever you create channel.
There can be a lot of channel members in the channel.
It's much better to lazy-load channel members whenever you need it.

It's important to deal with it immediately rather than a follow-up PR, because this makes it impossible to deploy that code in `master`. And also could become more and more complicated to fix, as any other code might rely on complete knowledge of channel members. So definitely worth doing more effort now to not pay the huge cost later on 😅